### PR TITLE
schema: add more documentation

### PR DESCRIPTION
**Pull request by oliver-sanders**
Partially addresses: #5612

There are a few folks poking around with the GraphQL schema which is great, but the documentation is a bit lacking.

Add a quick description to the most commonly used fields.

* Mark a couple of fields as obsolete/deprecated, see #5610
* Mark a couple of fields I couldn't provide descriptions for with TODO's (#5612).

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/network/schema.py [hunk 1]
```diff
@@ -606,101 +606,185 @@ class Workflow(ObjectType):
...
+    )
+    is_held_total = Int(
+        description='The number of "held" tasks.',
+    )
+    is_queued_total = Int(
+        description='The number of queued tasks.',
+    )
+    is_runahead_total = Int(
+        description=sstrip('''
+            The number of tasks which are held back by the runahead limmit.
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:730]:
```suggestion
            The number of tasks which are held back by the runahead limit.
```

File: cylc/flow/network/schema.py [hunk 2]
```diff
@@ -757,53 +841,108 @@ class Meta:
...
 
 class Task(ObjectType):
     class Meta:
-        description = """Task definition, static fields"""
+        description = sstrip("""
+            Task definitions.
+
+            These are the task "definitions" as they appear in the
+            configuration as opposed to task "instances" which you will find
+            in the `TaskProxies` field.
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:914]: Perhaps
```suggestion
            configuration as opposed to dynamic task "instances" which you
            will find in the `TaskProxies` field.
```

File: cylc/flow/network/schema.py [hunk 3]
```diff
@@ -885,24 +1024,60 @@ class Meta:
...
+        description='The inheritance order for this task.',
+    )
+    is_held = Boolean(
+        description='True if this task is "held".',
+    )
+    is_queued = Boolean(
+        description=sstrip('''
+            True if this task is "queued".
+
+            This relatetes to Cylc's internal task queues, not a job runner
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:1056]:
```suggestion
            This relates to Cylc's internal task queues, not a job runner
```

File: cylc/flow/network/schema.py [hunk 4]
```diff
@@ -913,57 +1088,64 @@ class Meta:
...
         strip_null=STRIP_NULL_DEFAULT,
         delta_store=DELTA_STORE_DEFAULT,
         delta_type=DELTA_TYPE_DEFAULT,
-        resolver=get_nodes_by_ids)
+        resolver=get_nodes_by_ids,
+    )
     first_parent = Field(
         lambda: FamilyProxy,
-        description="""Task first parent.""",
+        description="Task first parent (i.e. it's containing family).",
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:1133]:
```suggestion
        description="The task's first parent (i.e. its containing family).",
```

**Review by MetRonnie — COMMENTED** [review]
👍

**Review by MetRonnie — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Going to go with single approval for these docs.

**Pull request merged**